apple-silicon-support/packages/linux-asahi/default.nix

```diff
@@ -15,7 +18,7 @@
        "CONFIG_${builtins.elemAt kv 0}=${builtins.elemAt kv 1}";
     in lib.strings.concatMapStringsSep "\n" perLine lines;
 
-  readConfig = configfile: import (localPkgs.runCommand "config.nix" { } ''
+  readConfig = configfile: import (runCommand "config.nix" { } ''
```

Comment:
This has definitely broken `hardware.asahi.buildPkgs`. I am pretty sure the above comment is wrong. The real issue is that the `config.nix` file does not make its way to the ISO. Thus, when the configuration is evaluated, the `runCommand` needs to be executed to get the complete source code, but `runCommand` expects to run on a `hardware.asahi.buildPkgs` system and fails.

I want to keep the `localPkgs` hack for now (we don't need `lib = localPkgs.lib` though) and fix it later. That option will still be broken as it was for flakes users before, but it will be fine for the install scenario.

Reply:
I think the solution is to replace the `"config.nix"` IFD with a pure nix regex parser. It seems like it'll be pretty straightforward, and it can be upstreamed to nixpkgs since nixpkgs also uses an IFD currently.

Reply:
opened #51

Reply:
I've added back the `localPkgs` workaround for now.